### DISCUSSION TITLE
How to running monika - prometheus with docker

### DISCUSSION THREAD

**Discussion by heriy89** [hyperjumptech/monika#485] [General]
Hello, I'm a newbie 😃.
I previously tried this script from the tutorial on [medium.com](https://medium.com/hyperjump-tech/collecting-monika-with-prometheus-9faa7d484a30), but I got an error "**unknown flag: --prometheus**". After I checked, in the docker command, there's no **--prometheus** command. How to running Monika - Prometheus with docker command? Is there any other solution? 😺  Thanks before.

Here's I run this command:
```
docker run --name monika \
    -p 3001:3001 \
    -v ${PWD}/monika.json:/config/monika.json \
    --prometheus 3001 \
    --detach hyperjump/monika:latest
```

**Comment by dennypradipta** [comment]
> Hello, I'm a newbie :grinning:. I previously tried this script from the tutorial on [medium.com](https://medium.com/hyperjump-tech/collecting-monika-with-prometheus-9faa7d484a30), but I got an error "**unknown flag: --prometheus**". After I checked, in the docker command, there's no **--prometheus** command. How to running Monika - Prometheus with docker command? Is there any other solution? smiley_cat Thanks before.
> 
> Here's I run this command:
> 
> ```
> docker run --name monika \
>     -p 3001:3001 \
>     -v ${PWD}/monika.json:/config/monika.json \
>     --prometheus 3001 \
>     --detach hyperjump/monika:latest
> ```

Hello @heriy89 :grinning: 

You placed the parameters wrong. Here is how to run Monika + Prometheus using Docker:

1. **Run in Foreground**
    The command:
    ```bash
    docker run --name <container_name>  \
        -v ${PWD}/<your_config_name>:/config/monika.yml \
        -it hyperjump/monika:latest \
        monika -c /config/monika.yml --prometheus <prometheus_port>
    ```
    
    Example:
    ```bash
    docker run --name monika_interactive  \
        -v ${PWD}/config_denny.yml:/config/monika.yml \
        -it hyperjump/monika:latest \
        monika -c /config/monika.yml --prometheus 3001
    ```

2. **Run in Background** 
    The command:
    ```bash
    docker run --name <container_name>  \
        -v ${PWD}/<your_config_name>:/config/monika.yml \
        -d hyperjump/monika:latest \
        monika -c /config/monika.yml --prometheus <prometheus_port>
    ```

    Example:
    ```bash
    docker run --name monika_interactive  \
        -v ${PWD}/config_denny.yml:/config/monika.yml \
        -d hyperjump/monika:latest \
        monika -c /config/monika.yml --prometheus 3001
    ```

You put all the Monika parameters in the last line (such as `--version`, `--prometheus`, `--flush`, etc.) And the only difference for running fore/background is the `-it` and `-d` flags. 

Make sure you already run the Prometheus. I hope it helps!

**Reply by heriy89** [reply]
Thanks a lot @dennypradipta , solved 😃 👍
